From f157dc756515eea54d5bc36b759bbe8727c772ef Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Mon, 26 Aug 2024 15:41:50 -0500
Subject: [PATCH] [TESTING] - EMMC04G-MK27 mmc quirk

20240717: Start: eMMC life_time: 0x01 0x01
20240722: 6.10.0-bone6: eMMC life_time: 0x01 0x01
20240808: Start: new board... failed 20240825..
20240826: Start: new board...

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/mmc/core/quirks.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/mmc/core/quirks.h b/drivers/mmc/core/quirks.h
index 92905fc46436..623974dde9e0 100644
--- a/drivers/mmc/core/quirks.h
+++ b/drivers/mmc/core/quirks.h
@@ -128,6 +128,17 @@ static const struct mmc_fixup __maybe_unused mmc_blk_fixups[] = {
 	MMC_FIXUP("M62704", CID_MANFID_KINGSTON, 0x0100, add_quirk_mmc,
 		  MMC_QUIRK_TRIM_BROKEN),
 
+	/*
+	 * Kingston EMMC04G-MK27 are showing fast wear rates
+	 * 2024???? : testing MMC_QUIRK_TRIM_BROKEN (eMMC died)
+	 * 20240708 : testing MMC_QUIRK_BROKEN_SD_DISCARD (eMMC failing)
+	 * 20240717 : testing MMC_QUIRK_TRIM_BROKEN | MMC_QUIRK_BROKEN_CACHE_FLUSH (eMMC died)
+	 * 20240808 : testing MMC_QUIRK_TRIM_BROKEN | MMC_QUIRK_BROKEN_SD_DISCARD | MMC_QUIRK_BROKEN_CACHE_FLUSH | MMC_QUIRK_ERASE_BROKEN (eMMC failing)
+	 * 20240826 : testing MMC_QUIRK_TRIM_BROKEN | MMC_QUIRK_BROKEN_SD_CACHE
+	 */
+	MMC_FIXUP("MK2704", CID_MANFID_KINGSTON, 0x0100, add_quirk_mmc,
+		  MMC_QUIRK_TRIM_BROKEN | MMC_QUIRK_BROKEN_SD_CACHE),
+
 	/*
 	 * Some SD cards reports discard support while they don't
 	 */
-- 
2.39.2

